test/common/EdgeModule.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace common
+{
+    using System;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public enum EdgeModuleStatus
+    {
+        Running,
+        Stopped
+    }
+
+    public class EdgeModule
+    {
+        protected string deviceId;
+        protected IotHub iotHub;
+
+        public string Id { get; }
+
+        public EdgeModule(string id, string deviceId, IotHub iotHub)
+        {
+            this.deviceId = deviceId;
+            this.iotHub = iotHub;
+            this.Id = id;
+        }
+
+        public Task WaitForStatusAsync(EdgeModuleStatus desired, CancellationToken token)
+        {
+            return EdgeModule.WaitForStatusAsync(new []{this}, desired, token);
+        }
+
+        public static Task WaitForStatusAsync(EdgeModule[] modules, EdgeModuleStatus desired, CancellationToken token)
```

Comment:
EdgeModule class should care both its own module; To handle multiple module, this method should move to EdgeDevice class.

Reply:
Again, your suggestion makes sense conceptually, especially if we were creating a production SDK that callers could use to represent the relationships between devices and modules. But this is about running tests, wrapping scripts, etc. I want to create relationships between the object that make it easier to think about, write, and maintain the tests, but I don't want to take that goal so far that the implementation becomes strained an awkward.

Reply:
understood.